source/logHandler.py

```diff
@@ -189,6 +195,52 @@ def exception(self, msg="", exc_info=True, **kwargs):
 			return
 		self._log(level, msg, (), exc_info=exc_info, **kwargs)
 
+	def markFragmentStart(self):
+		"""Mark the current end of the log file as the start position of a
+		fragment to be later retrieved by L{getFragment}.
+		@returns: Whether a log file is in use and a position could be marked
+		@rtype: bool
+		"""
+		if (
+			not globalVars.appArgs
+			or globalVars.appArgs.secure
+			or not globalVars.appArgs.logFileName
+			or not self.handlers
+			or not isinstance(self.handlers[0], FileHandler)
+		):
+			return False
+		import codecs
+		with codecs.open(globalVars.appArgs.logFileName, "r", encoding="UTF-8") as f:
+			# io.IOBase.seek: whence=2 -- end of stream
+			f.seek(0, whence=2)
+			self.fragmentStart = f.tell()
+			return True
+
+	def getFragment(self):
+		"""Retrieve a fragment of the log starting from the position marked using
+		L{markFragmentStart}.
+		If L{fragmentStart} does not point to the current end of the log file, it
+		is reset to C{None} after reading the fragment.
+		@returns: The text of the fragment, or C{None} if L{fragmentStart} is None.
+		@rtype: str
+		"""
+		if (
+			self.fragmentStart is None
+			or not globalVars.appArgs
+			or globalVars.appArgs.secure
+			or not globalVars.appArgs.logFileName
+			or not self.handlers
+			or not isinstance(self.handlers[0], FileHandler)
+		):
+			return None
+		import codecs

```

Comment:
Is there any advantages to using codecs here instead of just open, or is it just because this PR has been created when NVDA was still running on Python 2?

Reply:
> or is it just because this PR has been created when NVDA was still running on Python 2?

Good catch, thanks.
Changed: c35bece